listener/src/event_handler/flux_transition_handler.rs

```diff
@@ -73,16 +74,30 @@ impl EventHandler for FluxTransitionHandler {
                     .store
                     .get_withdraws_of_slot(&slot)
                     .map_err(|e| failure::err_msg(format!("{}", e)))?;
-                account_state.apply_withdraws(&withdraws);
+                let valid_withdraws = account_state.apply_withdraws(&withdraws);
+                updated_records = valid_withdraws
```

Comment:
In future refactorings, I might be more confident about the correctness of this logic part, if we have a unit test for this logic